src/index.ts

```diff
@@ -138,95 +148,390 @@ export = (app: Probot) => {
+        const issueItemID = await GraphQL.getIssueItemID(projectID, issueNumber);
+        const issueItemStatus = await GraphQL.getIssueItemStatus(projectID, issueNumber);
+
+        if (issueItemStatus != "👀 In review") {
+          throw new Error("Issue is not in 👀 In review status. Can't change status to 🏗 In progress.");
+        } else {
+          // Change status to "🏗 In progress"
+          GraphQL.changeItemStatus(projectID, issueItemID, "🏗 In progress");
+        }
+      } else {
+        throw new Error('Previous comment is undefined. Issue remains in "👀 In review" status.');
+      }
+    } else if (linksToIssue(comment) && linksToIssue(previousComment)) {
+      // PR potetially linked to another issue
+      if (previousComment) {
+        const issueNumber = await GraphQL.getLinkedIssueNumber(projectID, pullRequestNumber, pullRequestRepo);
+        const previousIssueNumber = parseInt(previousComment?.split("#")[1].split(" ")[0]);
+
+        if (issueNumber == previousIssueNumber) {
+          // PR still linked to the same issue
+          console.log("PR still linked to the same issue. Nothing to do.");
+        } else {
+          // PR linked to another issue
+          // Change status of new linked issue to "👀 In review"
+          const issueItemID = await GraphQL.getIssueItemID(projectID, issueNumber);
+          const issueItemStatus = await GraphQL.getIssueItemStatus(projectID, issueNumber);
+
+          if (issueItemStatus != "🏗 In progress") {
+            throw new Error('Issue is not in "🏗 In progress" status. Can\'t change status to "👀 In review".');
+          } else {
+            GraphQL.changeItemStatus(projectID, issueItemID, "👀 In review");
+          }
+
+          // Change status of previous linked issue to "🏗 In progress"
+          const previousIssueItemID = await GraphQL.getIssueItemID(projectID, previousIssueNumber);
+          const previousIssueItemStatus = await GraphQL.getIssueItemStatus(projectID, previousIssueNumber);
+
+          if (previousIssueItemStatus != "👀 In review") {
+            throw new Error("Previous issue is not in 👀 In review status. Can't change status to 🏗 In progress.");
+          } else {
+            GraphQL.changeItemStatus(projectID, previousIssueItemID, "🏗 In progress");
+          }
+        }
+      } else {
+        throw new Error("Previous comment is undefined. Issue remains in 👀 In review status.");
+      }
+    } else {
+      throw new Error("PR is not linked to an issue.");
+    }
+  });
+
+  // For “👀 In review” issues, change status to “🏗 In progress” when PR is closed without merging
+  // For “👀 In review” issues, change status to “Done” when PR is merged
+  app.on("pull_request.closed", async (context) => {
+    const comment = context.payload.pull_request.body?.trim();
+    const pullRequestNumber = context.payload.pull_request.number;
+    const pullRequestRepo = context.payload.pull_request.head.repo.name;
+
+    if (linksToIssue(comment)) {
```

Comment:
i think it also should be checked for each new comment in any pr (or is it covered by .edited?). are you sure there is no way to get a list of linked prs from the issue?

Reply:
looks like smth like this can help:
```
query {
  repository(owner: "OWNER_NAME", name: "REPO_NAME") {
    issue(number: ISSUE_NUMBER) {
      timelineItems(itemTypes: [CONNECTED_EVENT]) {
        nodes {
          ... on ConnectedEvent {
            subject {
              ... on PullRequest {
                number
                title
                url
              }
            }
          }
        }
      }
    }
  }
}
```